### ISSUE TITLE
Selfhosting

### ISSUE THREAD

**Issue by Fabricio20** [skewten/PYE#7]
Hey Ivan,

For the past several years I've been redirecting to your pye.sq10.net for users wishing to import playlists into youtube from a project unrelated to PlugDJ, recently though, looks like you took down the website. 

I'm interested in self-hosting it, however when installing the dependencies it's erroring out with a missing middleware dependency of yours `github.com/seapunk/lisa-middleware.git`, is there any chance you could either get that up again or provide some sort of pre-compiled package?

Thanks!

**Comment by SEAPUNK** [comment]
`lisa-middleware` was a (now deleted) dependency for the dev server of PYE - you should be able to compile the .jade (https://jade-lang.com/), .scss (https://sass-lang.com/), and .ls (https://livescript.net/) files into .html, .css, and .js files (respectively) and serve it as static content

im surprised this project doesnt have a script that does that already, but prs are welcome to do just that

**Comment by SEAPUNK** [comment]
![2021-10-31_15-43-34](https://user-images.githubusercontent.com/4400726/139600609-57d195ec-c0e4-4eed-8c42-479580727b32.png)

for reference, this is what the file structure looks like on my server that served the pages, i dont remember what exact steps i took, but i imagine i did something similarly manual

**Comment by Fabricio20** [comment]
Thanks! I've managed to compile the project and after following the API keys guide you posted in #1 it seems to be working (besides me exceeding the API quota instantly during my testing.. and google asking for my entire family information to just create playlists..).

I've uploaded a simple npm script on my fork that builds the project for future reference, however I've removed all other dependencies besides `sass`, `livescript` and `jade` as they require python2 and no longer compile on modern linux and nodejs, which in turn means the devserver and gulpfile are no longer usable.

If you are interested I can create a pull request (with the gulpfile/devserver removed - I'm sadly not that well versed in the JS ecosystem) and a brief guide in the README for how to compile and configure the site.

Anyway, thanks for the quick reply and for keeping the project up for all those years!

**Comment by SEAPUNK** [comment]
> (besides me exceeding the API quota instantly during my testing.. and google asking for my entire family information to just create playlists..).

yeah thats the new google thing, its half the reason why i stopped bothering

yeah feel free to open whichever pr you want, ideally its some simple script that compiles the entire project into those output files without too many problems
